packages/flutter_tools/lib/src/commands/devices.dart

```diff
@@ -12,12 +12,34 @@ import '../globals.dart' as globals;
 import '../runner/flutter_command.dart';
 
 class DevicesCommand extends FlutterCommand {
+  DevicesCommand() {
+    argParser.addOption(
+      'timeout',
+      abbr: 't',
+      defaultsTo: null,
+      help: 'Time in seconds to wait for devices to attach. Longer timeouts may be necessary for networked devices.'
```

Comment:
Consider adding support for a `timeout` parameter to the daemon protocol in this PR as well: https://github.com/flutter/flutter/blob/master/packages/flutter_tools/doc/daemon.md#devicegetdevices

Reply:
Added the param and the documentation in the markup.